### PR TITLE
Set cudaStream for gpu resource

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

File: include/knowhere/gpu/gpu_res_mgr.h [hunk 1]
```diff
@@ -79,10 +80,12 @@ class GPUResMgr {
         if (!init_) {
             for (int64_t i = 0; i < gpu_params_.res_num_; ++i) {
                 auto gpu_res = new faiss::gpu::StandardGpuResources();
+                auto res = std::make_shared<Resource>(gpu_id_, gpu_res);
+                cudaStreamCreate(&res->stream_);
```
Comment (Presburger) [inline include/knowhere/gpu/gpu_res_mgr.h:84]: add a return status check.

Comment (cydrain) [reply include/knowhere/gpu/gpu_res_mgr.h]: done

**Pull request merged**